modules/agent/pkg/deployer/monitor.go

```diff
@@ -20,12 +31,144 @@ type DeploymentStatus struct {
 	ModifiedStatus []fleet.ModifiedStatus `json:"modifiedStatus,omitempty"`
 }
 
-func (m *Manager) plan(bd *fleet.BundleDeployment, ns string, objs ...runtime.Object) (apply.Plan, error) {
+func (m *Manager) plan(ctx context.Context, bd *fleet.BundleDeployment, ns string, objs ...runtime.Object) (apply.Plan, error) {
 	a, err := m.getApply(bd, ns)
 	if err != nil {
 		return apply.Plan{}, err
 	}
-	return a.DryRun(objs...)
+	plan, err := a.DryRun(objs...)
+	if err != nil {
+		return apply.Plan{}, err
+	}
+
+	result := objectset.ObjectByGVK{}
+	for _, obj := range objs {
+		result.Add(obj)
+	}
+
+	norms, err := m.normalizers(bd)
+	if err != nil {
+		return plan, err
+	}
+
+	for gvk, objs := range plan.Update {
+		client := m.dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    gvk.Group,
+			Version:  gvk.Version,
+			Resource: name.GuessPluralName(strings.ToLower(gvk.Kind)),
+		})
+
+		apiResources, err := m.discovery.ServerResourcesForGroupVersion(gvk.GroupVersion().String())
+		if err != nil {
+			return plan, err
+		}
+		var isNamespaced bool
+		for _, r := range apiResources.APIResources {
+			if r.Kind != gvk.Kind {
+				continue
+			}
+			if r.Namespaced {
+				isNamespaced = true
+			}
+		}
+		var errs []error
+		if isNamespaced {
+			for k, v := range result[gvk] {
```

Comment:
Both if and else do that same loop, you can make this code more concise

Reply:
I'm lost on what you are doing here. Let me look back at this in a bit.

Reply:
This two are not the same. This is to normalize the namespace field to have defaultNamespace or no namespace(if clusterScoped)